platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/EventMetadata.java

```diff
@@ -0,0 +1,273 @@
+     * @return the ID of the node that created this event
+     */
+    @NonNull
+    public NodeId getCreatorId() {
+        return creatorId;
+    }
+
+    /**
+     * Get the event descriptor for the self parent.
+     *
+     * @return the event descriptor for the self parent
+     */
+    @Nullable
+    public EventDescriptorWrapper getSelfParent() {
+        return selfParent;
+    }
+
+    /**
+     * Get the hash of the self parent.
+     *
+     * @return the hash of the self parent
+     */
+    @Nullable
+    public Hash getSelfParentHash() {
+        if (selfParent == null) {
+            return null;
+        }
+        return selfParent.hash();
+    }
+
+    /**
+     * Get the event descriptors for the other parents.
+     *
+     * @return the event descriptors for the other parents
+     */
+    @NonNull
+    public List<EventDescriptorWrapper> getOtherParents() {
+        return otherParents;
+    }
+
+    /**
+     * @return the hash of the other parent
+     */
+    @Nullable
+    @Deprecated
+    public Hash getOtherParentHash() {
+        if (otherParents == null || otherParents.isEmpty()) {
+            return null;
+        }
+        if (otherParents.size() == 1) {
+            return otherParents.getFirst().hash();
+        }
+        // 0.46.0 adds support for multiple other parents in the serialization scheme, but not yet in the
+        // implementation. This exception should never be reached unless we have multiple parents and need to
+        // update the implementation.
+        throw new UnsupportedOperationException("Multiple other parents is not supported yet");
+    }
+
+    /**
+     * Check if the event has other parents.
+     *
+     * @return true if the event has other parents
+     */
+    public boolean hasOtherParent() {
+        return otherParents != null && !otherParents.isEmpty();
+    }
+
+    /** @return a list of all parents, self parent (if any), + all other parents */
+    @NonNull
+    public List<EventDescriptorWrapper> getAllParents() {
+        return allParents;
+    }
+
+    @NonNull
+    public Instant getTimeCreated() {
+        return timeCreated;
+    }
+
+    /**
+     * @return list of transactions wrappers
+     */
+    @NonNull
+    public List<TransactionWrapper> getTransactions() {
+        return transactions;
+    }
+
+    public long getGeneration() {
+        return generation;
+    }
+
+    /**
+     * Get the event descriptor for this event, creating one if it hasn't yet been created. If called more than once
+     * then return the same instance.
+     *
+     * @return an event descriptor for this event
+     * @throws IllegalStateException if called prior to this event being hashed
+     */
+    @NonNull
+    public EventDescriptorWrapper getDescriptor(final long birthRound) {
+        if (descriptor == null) {
+            if (getHash() == null) {
+                throw new IllegalStateException("The hash of the event must be set before creating the descriptor");
```

Comment:
probably should add a unit test here that expects an illegal state exception so that we know if this is accidentally removed.

Reply:
added